Sources/StreamChat/Utils/StateBuilder.swift

```diff
@@ -0,0 +1,24 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// A builder for objects requiring @MainActor.
+final class StateBuilder<State> {
```

Comment:
I added this helper class for enabling creating state objects on the main actor.

The flow is that you create this in non-isolated context:
```swift
stateBuilder = StateBuilder {
  environment.blabla
```

And it is triggered when we access `.state` for the first time and `.state` requires @MainActor. 
`@MainActor public lazy var state: MessageSearchState = stateBuilder.build()`

This was the only good way to workaround the case of, for example,
ChatState is @MainActor but its owning type, Chat, is non-isolated.